src/core/controllers/base.go

```diff
@@ -54,10 +57,43 @@ type messageDetail struct {
 	UUID string
 }
 
+func redirectForOIDC(ctx context.Context, username string) bool {
+	am, _ := ctx.Value(filter.AuthModeKey).(string)
+	if am != common.OIDCAuth {
+		return false
+	}
+	u, err := dao.GetUser(models.User{Username: username})
+	if err != nil {
+		log.Warningf("Failed to get user by name: %s, error: %v", username, err)
+	}
+	if u == nil {
+		return true
+	}
+	ou, err := dao.GetOIDCUserByUserID(u.UserID)
+	if err != nil {
+		log.Warningf("Failed to get OIDC user info for user, id: %d, error: %v", u.UserID, err)
+	}
+	if ou != nil {
+		return true
+	}
+	return false
+}
+
 // Login handles login request from UI.
 func (cc *CommonController) Login() {
 	principal := cc.GetString("principal")
 	password := cc.GetString("password")
+	if redirectForOIDC(cc.Ctx.Request.Context(), principal) {
```

Comment:
am, _ := ctx.Value(filter.AuthModeKey).(string)
if am != common.OIDCAuth {
	if redirectForOIDC(cc.Ctx.Request.Context(), principal) {
               ........
        }
}

Reply:
That was checked in `redirectForOIDC `